### PR TITLE
On the fly WP Mission Planner

### LINKED ISSUE #4663: Adding waypoints in flight
## Current Behavior
Current behavior is not available. Its a new feature.

## Desired Behavior
Making a waypoint mission on a tablet with an app like Mission Planner is a lot of fun, but sometimes beeing at the location gives you new insight in how to make a new mission. This would be especially a nice feature for filming with a drone. this way you can determine your filming locations and then execute the whole waypoint mission.

## Suggested Solution
When at the flying site, manouver the drone at the desired location, set poshold and flick a switch to set waypoint. Then continue to the next point, set waypoint and so on. Waypoints have to be stored in flight. 
When landed and disarmed waypointroute has to be saved to eeprom and set as current mission, disarming means mission had ended and can be saved. When taking off again, waypoint function executes the mission. When in poshold and the switch "set waypoint" is flicked again the previous route is deleted and all starts over again. 

If this all is possible, maybe it would be nice if the new route can be downloaded to an app to store it in a file for later flights

## Who does this impact? Who is this for?
All users

### PR DESCRIPTION
**Pull request by breadoven**
Closes https://github.com/iNavFlight/inav/issues/4663.

PR provides a mode for setting WPs at the current location while navigating. WPs are stored in FC memory in the normal mission format so the mission can be flown when the mode is off without having to land to to save and reload. All WPs are the basic WP type (action 1) with altitude saved using the absolute datum. WPs can be saved using the current stick command after disarm.

When the mode is active it uses the WP mode switch to set a WP. WP mode is disabled when mission planner mode is active and mission planner mode disabled when WP mode is active (including WP RTH fall back). All other flight modes are unaffected, normal RTH included (only WP fall back RTH is disabled).

Feed back is provided by a Mission field in the OSD that shows the number of WPs loaded and the current status. A system message is also displayed showing the mode is active.

Only option is Reset which allows mission WP planning to be restarted from the first WP by toggling the WP mode switch ON-OFF-ON (within 1s).

Currently the mode can be used when Disarmed so EEPROM mission loading is disabled if WPs have been loaded using mission planner mode. Prevents stored WP missions overwriting mission planner WPs.

Not yet tested in flight, only fly by hand "flight" tests ... which worked fine.

### CODE REVIEW

File: src/main/navigation/navigation.c [hunk 1]
```diff
@@ -114,6 +114,7 @@ PG_RESET_TEMPLATE(navConfig_t, navConfig,
             .rth_alt_control_override = SETTING_NAV_RTH_ALT_CONTROL_OVERRIDE_DEFAULT, // Override RTH Altitude and Climb First using Pitch and Roll stick
             .nav_overrides_motor_stop = SETTING_NAV_OVERRIDES_MOTOR_STOP_DEFAULT,
             .safehome_usage_mode = SETTING_SAFEHOME_USAGE_MODE_DEFAULT,
+            .mission_planner_resume = SETTING_NAV_MISSION_PLANNER_RESUME_DEFAULT,     // resume WP entry when Mission Planner mode is restarted
```
Comment (stronnag) [inline src/main/navigation/navigation.c]: Please update PG version (12 => 13). We all forget this.

Comment (breadoven) [reply src/main/navigation/navigation.c]: Done ...

**Review by DzikuVx — APPROVED** [review]
lgtm!

**Comment by DzikuVx** [comment]
@breadoven resolve conflict please

**Comment by breadoven** [comment]
Need to double check Multi Missions don't cause a problem with this when disarmed. OK to merge otherwise.

**Comment by shannonbaker** [comment]
Hi, can someone please provide simple steps to show how this should work?  The main description is confusing.  Thanks!

**Comment by breadoven** [comment]
> Hi, can someone please provide simple steps to show how this should work? The main description is confusing. Thanks!

Any use ?

https://github.com/iNavFlight/inav/wiki/Navigation-modes#wp-planner---on-the-fly-waypoint-mission-planner

**Comment by Todor319** [comment]
hi how can i get wp point data on my rc via ibus telemetri

**Comment by b14ckyy** [comment]
> hi how can i get wp point data on my rc via ibus telemetri

Please ask for support in the discussions area or the official support channels on FB or Discord.

**Pull request merged**